src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs

```diff
@@ -1022,10 +1022,26 @@ private void ProcessStartOfFrameMarker(BufferedReadStream stream, int remaining,
             int index = 0;
             for (int i = 0; i < componentCount; i++)
             {
+                // 1 byte: component identifier
+                byte componentId = this.temp[index];
+
+                // 1 byte: component sampling factors
                 byte hv = this.temp[index + 1];
                 int h = (hv >> 4) & 15;
                 int v = hv & 15;
 
+                // Validate: 1-4 range
+                if (h is < 1 or > 4)
+                {
+                    JpegThrowHelper.ThrowInvalidImageContentException($"Bad horizontal sampling factor: {h}");
+                }
+
+                // Validate: 1-4 range
+                if (v is < 1 or > 4)
+                {
+                    JpegThrowHelper.ThrowInvalidImageContentException($"Bad vertical sampling factor: {v}");
+                }
```

Comment:
Wish we had an easy way to construct tests that can exercise all the error cases.

Reply:
We can expose this marker parsing methods as static ones and them separately which is good enough IMO. Somewhat automatic fuzzer would also be good but too much work for now, working on decoder optimizations for now which also need a lot of tests.

Reply:
Testing static methods against constructed binary data is also a non trivial effort, but might worth investigating eventually.

One more thing for this PR:
Some of the coverage change may indicate that unused methods, if so, we should delete them:
https://app.codecov.io/gh/SixLabors/ImageSharp/compare/1899/changes

Reply:
There actually were 2 unused methods, removed them. I've also introduced a new helper throw method to remove those doubled hand-written sampling exceptions.